### PR TITLE
Fix 502 error on Yii 3 progress page

### LINKED ISSUE #965: yii3-progress page giving error 502
https://www.yiiframework.com/yii3-progress

image

### PR DESCRIPTION
**Pull request by arogachev**
Fixes #965.

- Refactoring.
- Added console command.
- Improved caching.
- Additionally added caching for "Release statuses" pages.
- Handle `Github\Exception\RuntimeException` (Something went wrong while executing your query. This may be the result of a timeout, or it could be a GitHub bug). Happens when executing GraphQL query.

Deploy instructions:

- Add `./yii github-progress` command to crontab. Interval - `@hourly`.

### CODE REVIEW

File: components/github/GithubProgress.php [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+use Github\Client as GithubClient;
+use Yii;
+use yii\base\InvalidArgumentException;
+use yii\base\InvalidConfigException;
+
+class GithubProgress
+{
+    const VERSIONS = ['1.1', '2.0', '3.0'];
+    const PACKAGES_CACHE_KEY_PREFIX = 'github_progress_packages';
```
Comment (arogachev) [inline components/github/GithubProgress.php:13]: Forgot to remove these.

Comment (arogachev) [reply components/github/GithubProgress.php:13]: :heavy_check_mark:

**Comment by samdark** [comment]
@arogachev https://github.com/yiisoft-contrib/yiiframework.com/blob/master/scripts/cron-hourly.sh

**Comment by arogachev** [comment]
@samdark Added. :ok_hand:

**Pull request merged**